### ISSUE TITLE
ENV Variables

### ISSUE THREAD

**Issue by McStuffins** [LoganMD/Crunchycord#2]
You have an env variable in your repo. Anyone can see it who views this repo. You should use `process.env` to access the env variables. I am referring to this:
https://github.com/LoganMD/Crunchycord/blob/71e653ae15980bc6c13bbf5156b9db8035551287/server/server.js#L2

**Comment by McStuffins** [comment]
I also think that you should use env variables for the port as well 👍

**Comment by vityavv** [comment]
This is not an env variable. This is the token that starts the rich presence, which is needed to show which rich presence you're using. If this token wasn't there, people using this app would have to make their own RPC application (with their own images) to make it work

**Comment by McStuffins** [comment]
Where is the token generated? I am pretty sure that it is a *client* id meaning the user?

**Comment by LoganMD** [comment]
it's using the discord RPC, that iD is from the application that has the icons

**Comment by LoganMD** [comment]
i'll consider allowing custom ports

**Comment by McStuffins** [comment]
That makes sense. Also, if you do not consider allowing custom ports, then your application could potentially not run because of the port constraints.

**Comment by LoganMD** [comment]
i'm aware, so im planning to add that. the only problem is i'm not sure how to set the port from the extension

**Comment by McStuffins** [comment]
I will do some research 👀

**Comment by LoganMD** [comment]
thx

**Comment by McStuffins** [comment]
Actually, I think that it is just set when running the extension. This is an NPM module?

**Comment by vityavv** [comment]
There is a chrome extension in here too, which sends requests to the server, which then sets the RPC.

**Comment by LoganMD** [comment]
yeah, im hoping to make it so only the chrome extension is needed in the future

**Comment by vityavv** [comment]
There is an example of discord rpc in the browser: https://discord.js.org/#/docs/rpc/master/general/welcome

The discord-rpc module is the module that discord-rich-presence wraps around, so switching wouldn't be too hard?

**Comment by LoganMD** [comment]
the only problem is the RPC is limited to devs who get permission first. idk if they would allow

**Comment by LoganMD** [comment]
OHHHH, i see. i'll try that i guess.

**Comment by AleanaA** [comment]
As a suggestion, wouldn't using Electron instead of a chrome extension be a viable solution as well? There's an issue where your status is removed if you close a tab that isn't crunchyroll, so that would most likely solve both issues.

**Comment by vityavv** [comment]
Can electron detect the fact that you're watching crunchyroll or would you have to put crunchyroll inside the electron app as well? If you had to put chrunchyroll inside of the app, what licensing troubles would happen?

**Comment by AleanaA** [comment]
All electron does is act like a web browser afaik, which shouldn't cause any issues in terms of licensing.

**Comment by McStuffins** [comment]
I guess that you would actually be able to use headless chrome without headless to navigate to crunchyroll and be able to do whatever you would like. That would be an easier solution, I believe.

**Comment by AleanaA** [comment]
Easier doesn't mean better. After using it for awhile there's other issues, like the status not being removed when chrome is closed. Imho a chrome extension works for this, but it shouldn't be an end all be all solution for this. This also allows users not using chrome to use this, as users using firefox, opera(Might still work, didn't test), vivaldi, edge, or safari aren't able to use this.

**Comment by AleanaA** [comment]
The point of using electron would be to not use chrome at all. The entire app would be be contained and wouldn't have to use a separate browser and a server like the current setup does.

**Comment by McStuffins** [comment]
I will have to look into how this could be possible.

**Comment by LoganMD** [comment]
that's pretty good, do you mind if i try using your code to improve mine? i know electron as well

**Comment by AleanaA** [comment]
Of course! My whole reason for making that repo was to give an easy reference if someone here wanted to improve this.

**Comment by LoganMD** [comment]
alright thanks, ill try to understand how your code works

**Comment by LoganMD** [comment]
I'm considering using https://github.com/GoogleChrome/puppeteer and electron for the next edition

**Comment by McStuffins** [comment]
Puppeteer is what I was talking about you using, but I think that since electron is essentially a browser, you will not need to do this. I will have to do more research on that though...

**Comment by LoganMD** [comment]
I just don't want to make crunchyroll an application, that's why I'm not making it load the site inside electron. Electron will be used as some sort of graphical interface

**Comment by McStuffins** [comment]
Hmm. I understand that for sure. I would like to see it as an auto-detected game some how. We will see if it it’s possible. I will do research when I have free time..

**Comment by McStuffins** [comment]
I think that you are just able to use electron with the `window.open` function
